dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/datavalue/hibernate/HibernateDataValueStore.java

```diff
@@ -330,18 +328,29 @@ else if ( params.hasStartEndDate() )
     {
         SqlHelper sqlHelper = new SqlHelper( true );
 
-        String orgUnitId = params.isReturnParentForOrganisationUnits() ? "opath.id" : "dv.sourceid";
+        if ( params.isIncludeChildrenForOrganisationUnits() )
```

Comment:
Could we move this check to the service layer potentially? Ideally validation should be in one place. We can also use an `ErrorCode` for programmatic handling.

Reply:
Thanks. Done!

Reply:
Overcome by events. In the later refactor to use OrganisationUnitSelectionMode, this error check was removed.